flink-batch-connectors/flink-avro/src/main/java/org/apache/flink/api/java/io/AvroInputFormat.java

```diff
@@ -119,12 +144,18 @@ public E nextRecord(E reuseValue) throws IOException {
 		if (reachedEnd()) {
 			return null;
 		}
-		
-		if (!reuseAvroValue) {
-			reuseValue = InstantiationUtil.instantiate(avroValueType, Object.class);
+		if (isGenericRecord) {
```

Comment:
Can we replace the nested-if with `return reuseAvroValue ? dataFileReader.next(reuseValue) : dataFileReader.next();`, remove the `isGenericRecord` field, and flatten the `GenericDatumReader` / `SpecificDatumReader` / `ReflectDatumReader` if-statement above?